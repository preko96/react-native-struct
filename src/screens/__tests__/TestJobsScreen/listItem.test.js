import React from 'react'
import ListItem from '../../TestJobsScreen/listItem'

let wrapper

const mockProps = {
	secId: '0',
	rowId: '0',
	rowMap: {
		'00': {
			props: {
				closeRow: jest.fn()
			}
		}
	},
	onPress: jest.fn(),
	top: 'topTitle',
	completedForms: 10,
	totalForms: 50,
}

beforeEach(()=>{
	jest.clearAllMocks()
	wrapper = shallow(<ListItem {...mockProps} />)
})

it('renders without crashing', () => {
	expect(wrapper).toMatchSnapshot()
})

it('close side and call the onPress callback', () => {
	wrapper.simulate('Press')
	expect(mockProps.rowMap['00'].props.closeRow).toHaveBeenCalledTimes(1)
	expect(mockProps.onPress).toHaveBeenCalledTimes(1)
}) 


