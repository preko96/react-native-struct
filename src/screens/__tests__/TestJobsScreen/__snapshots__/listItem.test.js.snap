// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<Styled(ListItem)
  icon={true}
  onPress={[Function]}
  style={
    Object {
      "backgroundColor": "white",
      "height": "auto",
      "marginLeft": 0,
    }
  }
>
  <Styled(Body)
    style={
      Object {
        "height": "auto",
        "paddingBottom": 8,
        "paddingLeft": 16,
        "paddingTop": 8,
      }
    }
  >
    <Styled(Text)
      numberOfLines={1}
      style={
        Object {
          "marginBottom": 4,
        }
      }
    >
      topTitle
    </Styled(Text)>
    <Styled(ViewNB)
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <Styled(ViewNB)
        style={
          Object {
            "backgroundColor": "red",
            "borderRadius": 50,
            "height": 14,
            "marginRight": 8,
            "width": 14,
          }
        }
      />
      <Styled(Text)
        numberOfLines={1}
        style={
          Object {
            "color": "#ababab",
            "fontSize": 13,
          }
        }
      >
        10
        /
        50
         Forms Completed
      </Styled(Text)>
    </Styled(ViewNB)>
  </Styled(Body)>
  <Styled(Right)
    style={
      Object {
        "height": "auto",
        "paddingBottom": 8,
        "paddingTop": 8,
      }
    }
  >
    <Styled(Icon)
      name="arrow-forward"
    />
  </Styled(Right)>
</Styled(ListItem)>
`;
